src/coreComponents/meshUtilities/CPMesh/CPMeshParser.cpp

```diff
@@ -0,0 +1,372 @@
+
+      nX = bufInt[0];
+      nY = bufInt[1];
+      nZ = bufInt[2];
+    }
+  }
+  GEOSX_THROW_IF( nX <= 0 || nY <= 0 || nZ <= 0,
+                  "Neither SPECGRID nor DIMENS was found",
+                  InputError );
+
+}
+
+void CPMeshParser::readMesh( Path const & filePath,
+                             CPMeshDimensions const & cPMeshDims )
+{
+  // Note: This is definitely not what we want to do (reading the full file into a string)
+  //       For now, to get a quick start, I ported this from PAMELA. I will improve that in a second step23
+  string fileContent = CPMeshStringUtilities::fileToString( filePath );
+  std::istringstream meshFile;
+  meshFile.str( fileContent );
+
+  bool foundCOORD = false;
+  bool foundZCORN = false;
+  bool foundACTNUM = false;
+  bool foundPERMX = false;
+  bool foundPERMY = false;
+  bool foundPERMZ = false;
+
+  std::string line;
+  while( getline( meshFile, line ) )
+  {
+    CPMeshStringUtilities::removeStringAndFollowingContentFromLine( "--", line );
+    CPMeshStringUtilities::removeExtraSpaces( line );
+    CPMeshStringUtilities::removeEndOfLine( line );
+    CPMeshStringUtilities::removeTab( line );
+    CPMeshStringUtilities::trim( line );
+
+    // TODO: shorten, there must be a way to factorize all these functions
+    //       at least, we shoud be able to merge readLocalACTNUM and readLocalPROP
+
+    // at this point, SPECGRID has already been read, we skip it
+    if( line == "COORD" )
+    {
+      foundCOORD = true;
+      readLocalCOORD( meshFile, cPMeshDims );
+    }
+    else if( line == "ZCORN" )
+    {
+      foundZCORN = true;
+      readLocalZCORN( meshFile, cPMeshDims );
+    }
+    else if( line == "ACTNUM" )
+    {
+      foundACTNUM = true;
+      readLocalACTNUM( meshFile, cPMeshDims );
+    }
+    else if( line == "PERMX" )
+    {
+      foundPERMX = true;
+      readLocalPROP( meshFile, cPMeshDims, m_permx );
+    }
+    else if( line == "PERMY" )
+    {
+      foundPERMY = true;
+      readLocalPROP( meshFile, cPMeshDims, m_permy );
+    }
+    else if( line == "PERMZ" )
+    {
+      foundPERMZ = true;
+      readLocalPROP( meshFile, cPMeshDims, m_permz );
+    }
+    else if( line == "PORO" )
+    {
+      readLocalPROP( meshFile, cPMeshDims, m_poro );
+    }
+  }
+
+  if( !foundACTNUM )
+  {
+    fillLocalACTNUM( cPMeshDims );
+  }
+
+  // TODO: error out if one of them is missing
+  if( foundPERMX && foundPERMY && foundPERMZ )
+  {
+    m_perm.resizeDimension< 0, 1 >( m_permx.size(), 3 );
+    for( localIndex i = 0; i < m_permx.size(); ++i )
+    {
+      m_perm( i, 0 ) = m_permx( i );
+      m_perm( i, 1 ) = m_permy( i );
+      m_perm( i, 2 ) = m_permz( i );
+    }
+  }
+
+  GEOSX_THROW_IF( !foundCOORD || !foundZCORN,
+                  "At least one of the following keywords was not found: COORD, ZCORN",
+                  InputError );
+
+
+}
+
+void CPMeshParser::readLocalCOORD( std::istringstream & meshFile,
```

Comment:
meshFile is a const?

Reply:
I tried to make it `const`, but I cannot because the `getline` function used in this function takes a non-const `std::istringstream`